crates/factor-wasi/src/lib.rs

```diff
@@ -0,0 +1,182 @@
+    type InstanceState = InstanceState;
+
+    fn init<Factors: SpinFactors>(&mut self, mut ctx: InitContext<Factors, Self>) -> Result<()> {
+        fn type_annotate<T, F>(f: F) -> F
+        where
+            F: Fn(&mut T) -> &mut dyn WasiView,
+        {
+            f
+        }
+        let get_data = ctx.get_data_fn();
+        let closure = type_annotate(move |data| get_data(data) as &mut dyn WasiView);
+        if let Some(linker) = ctx.linker() {
+            use wasmtime_wasi::bindings;
+            bindings::clocks::wall_clock::add_to_linker_get_host(linker, closure)?;
+            bindings::clocks::monotonic_clock::add_to_linker_get_host(linker, closure)?;
+            bindings::filesystem::types::add_to_linker_get_host(linker, closure)?;
+            bindings::filesystem::preopens::add_to_linker_get_host(linker, closure)?;
+            bindings::io::error::add_to_linker_get_host(linker, closure)?;
+            bindings::io::poll::add_to_linker_get_host(linker, closure)?;
+            bindings::io::streams::add_to_linker_get_host(linker, closure)?;
+            bindings::random::random::add_to_linker_get_host(linker, closure)?;
+            bindings::random::insecure::add_to_linker_get_host(linker, closure)?;
+            bindings::random::insecure_seed::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::exit::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::environment::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::stdin::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::stdout::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::stderr::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::terminal_input::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::terminal_output::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::terminal_stdin::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::terminal_stdout::add_to_linker_get_host(linker, closure)?;
+            bindings::cli::terminal_stderr::add_to_linker_get_host(linker, closure)?;
+            bindings::sockets::tcp::add_to_linker_get_host(linker, closure)?;
+            bindings::sockets::tcp_create_socket::add_to_linker_get_host(linker, closure)?;
+            bindings::sockets::udp::add_to_linker_get_host(linker, closure)?;
+            bindings::sockets::udp_create_socket::add_to_linker_get_host(linker, closure)?;
+            bindings::sockets::instance_network::add_to_linker_get_host(linker, closure)?;
+            bindings::sockets::network::add_to_linker_get_host(linker, closure)?;
+            bindings::sockets::ip_name_lookup::add_to_linker_get_host(linker, closure)?;
+        }
+        Ok(())
+    }
+}
+
+pub trait FilesMounter {
+    fn mount_files(&self, app_component: &AppComponent, ctx: MountFilesContext) -> Result<()>;
+}
+
+pub struct DummyFilesMounter;
```

Comment:
I'm confused. Is this meant to be a temporary thing before this gets merged?

Reply:
Yes and probably for unit tests that don't care about files.

Reply:
Oh also the actual FilesMounter would be defined by the runtime, though perhaps `spin up`'s mounter will live here as a standard impl.